modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/toolbarbuttons/plugin.js

```diff
@@ -50,22 +225,75 @@
 			});
 
 			editor.ui.addRichCombo('TableHeaders', {
+				_headersLabels: {
+					[HEADING_BOTH]: `${editor.lang.table.headers}: ${editor.lang.table.headersBoth}`,
+					[HEADING_COL]: `${editor.lang.table.headers}: ${editor.lang.table.headersColumn}`,
+					[HEADING_NONE]: `${editor.lang.table.headers}: ${editor.lang.table.headersNone}`,
+					[HEADING_ROW]: `${editor.lang.table.headers}: ${editor.lang.table.headersRow}`,
+				},
+
+				_setHeading(heading, editor) {
+					setHeading({editor, heading, table: null});
+
+					this.setValue(heading, this._headersLabels[heading]);
+				},
+
 				init() {
-					const headersPrefix = editor.lang.table.headers;
+					this.add(
+						HEADING_NONE,
+						this._headersLabels[HEADING_NONE],
+						this._headersLabels[HEADING_NONE]
+					);
+					this.add(
+						HEADING_ROW,
+						this._headersLabels[HEADING_ROW],
+						this._headersLabels[HEADING_ROW]
+					);
+					this.add(
+						HEADING_COL,
+						this._headersLabels[HEADING_COL],
+						this._headersLabels[HEADING_COL]
+					);
+					this.add(
+						HEADING_BOTH,
+						this._headersLabels[HEADING_BOTH],
+						this._headersLabels[HEADING_BOTH]
+					);
 
-					const headersNone = `${headersPrefix}: ${editor.lang.table.headersNone}`;
-					const headersRow = `${headersPrefix}: ${editor.lang.table.headersRow}`;
-					const headersColumn = `${headersPrefix}: ${editor.lang.table.headersColumn}`;
-					const headersBoth = `${headersPrefix}: ${editor.lang.table.headersBoth}`;
+					const currentHeading = getHeading({editor, table: null});
 
-					this.add(headersNone, headersNone, headersNone);
-					this.add(headersRow, headersRow, headersRow);
-					this.add(headersColumn, headersColumn, headersColumn);
-					this.add(headersBoth, headersBoth, headersBoth);
+					this._setHeading(currentHeading, editor);
```

Comment:
This is fine, but we are missing one bit of behavior. `init` is invoked on toolbar click, when the panel in first opened. Before this, on initial table creation, toolbar button label is set to `Heading`. This is because we set
```js
label: editor.lang.table.headers,
```
Instead of this, we should set it the same way as in `init`, based on `getCurrentHeading`.

Reply:
It's not possible set a label reusing getCurrentHeading function because `init: ` part is being read before the editor is ready and I searched for something like `editor.instanceReady` or something like this and I didn't find for adding a conditional to add a placeholder value when editor is not on the screen. See:

<img width="1917" alt="Screen Shot 2022-02-23 at 15 50 00" src="https://user-images.githubusercontent.com/7663513/155387202-848ab074-45cb-4bc3-aba2-935df5175434.png">

Reply:
When step into the function, we don't have a selection

<img width="499" alt="Screen Shot 2022-02-23 at 15 52 11" src="https://user-images.githubusercontent.com/7663513/155387530-c3b73695-2ad7-449c-9b74-947192a6ced7.png">

Reply:
So, I'll send a version without this proposal

Reply:
OK, let's call this good enough :+1:  It's too much complications for a very minor improvement. We may return to this in the future, if it's reported as a bug.